go/arrow/schema.go

```diff
@@ -197,6 +197,26 @@ func (sc *Schema) FieldIndices(n string) []int {
 	return sc.index[n]
 }
 
+// FieldIndex return index of the unique field with the given name.
+// Returns -1 if the name isn’t found or there are several fields with the given name.
+func (sc *Schema) FieldIndex(name string) int {
+	indices := sc.index[name]
+	if len(indices) != 1 {
+		return -1
+	}
+	return indices[0]
+}
+
+func (sc *Schema) InsertField(i int, f Field) *Schema {
```

Comment:
I think this might be a bit confusing, because the name `InsertField` suggests that the field will be insert in-place, but in fact an entirely new schema is created and returned. If we have an `InsertField` function at all, I think it should not have a return value and do the insertion in-place.

We can also get away with not having this function at all by having clients do something like:

```
oldFields := sch.Fields()
newFields := make([]arrow.Field, len(oldFields))
copy(newFields[:i],  oldFields[:i])
copy(newFields[i+1:],  oldFields[i:])
newSchema := NewSchema(newFields, sch.Metadata())
```

(which they can wrap in a function if needs be)